File: _tests/java/src/agshmne/Personnage.java
package agshmne;

import java.util.HashMap;
import java.util.Map;

public class Personnage {
	private Map<String, Integer> vital = new HashMap<>();
	private int position;

	public Personnage(){	
		vital.put("vie", 0);
		vital.put("mana", 0);
		position = 0;
		
	}
	
	public Integer getVie(){	
		return vital.get("vie");
	}
	
	
}

File: _tests/ruby/personnage.rb
require_relative 'position'

<<-DOC
Définit tout ce qui est en rapport avec le Personnage
DOC

# Personnage : Définit le personnage
#
class Personnage
    private
    def initialize()
        @vital = {'vie'=>0,'mana'=>0}
        @carac = {'puissance'=>0,'defense'=>0}
        @position = Position.new
    end
    public
    def go_to(direction)
        @position.move(direction)
    end
    def get_x()
        @position.get_x
    end
    def get_y()
        @position.get_y
    end
end

